frontend/src/components/common/navigation/RouterTabs.tsx

```diff
@@ -35,22 +44,24 @@ const RouterTabs = ({ routes }: RouterTabsProps) => {
       <Tabs
         index={currentRouteIndex}
         isLazy
-        onChange={(index) =>
-          history.push({
-            pathname: generatePath(routes[index].path, matchParams),
+        onChange={(index) => {
+          const newPath = generatePath(routes[index].path, matchParams);
+          if (newPath === location.pathname) return;
+          return history.push({
+            pathname: newPath,
             state: location.state,
-          })
-        }
+          });
+        }}
       >
         <TabList>
           {routes.map(({ name, path }) =>
             name ? <Tab key={path}>{name}</Tab> : null,
           )}
         </TabList>
         <TabPanels>
-          {routes.map(({ path, Component }) => (
+          {routes.map(({ path, Component, element }) => (
             <TabPanel key={path} p={0} pt={8}>
-              <Component />
+              {Component ? <Component /> : element}
```

Comment:
A component is a `() => React.ReactNode`! I wanted to support `element` because it makes it slightly cleaner to pass a `<RedirectTo ... />` rather than `() => <RedirectTo ... />` and similar.

Reply:
A component is a `() => React.ReactNode`! I wanted to support `element` because it makes it slightly cleaner to pass a `<RedirectTo ... />` rather than `() => <RedirectTo ... />` and similar.

Reply:
In general a lot of libraries like `react-router` also seem to be migrating to an `element` style instead of `Component`!